pkg/cloudprovider/providers/gce/gce_kms.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```

Comment:
This file is structured so as to work on and off GCP, both.
There is a check for the cloud, in the `newGoogleKMSService` function, which decides to look for local credentials if the apiserver is not on GCE/GKE.

It would be possible to run an apiserver on a non-GCE VM, and still access KMS.

This file itself is a stand-alone file, and can be kept anywhere in cloudprovider.

Reasons I kept this file here:
* All Google-specific code in one directory.
* This file uses `gce/token_source.go` to authenticate to Google services.
* Not adding any new package to cloudproviders